packages/service/ni_measurement_plugin_sdk_service/pin_map/_client.py

```diff
@@ -72,9 +80,10 @@ def update_pin_map(self, pin_map_path: Union[str, pathlib.Path]) -> str:
             The resource id of the pin map that is registered to the pin map service.
         """
         # By convention, the pin map id is the .pinmap file path.
+        with open(pin_map_path, "r", encoding="utf-8-sig") as file:
+            xml_content = file.read()
         request = pin_map_service_pb2.UpdatePinMapFromXmlRequest(
-            pin_map_id=str(pin_map_path),
-            pin_map_xml=pathlib.Path(pin_map_path).read_text(encoding="utf-8"),
+            pin_map_id=str(pin_map_path), pin_map_xml=xml_content
```

Comment:
Why replace read_text with open+read? You can pass utf-8-sig to read_text.

Reply:
I think I overlooked this case. I went with context manager purely for resources management. But seems like `pathlib.path` internally handles the file open and close operations. Hence, I have reverted back the context manager change.